src/builder.rs

```diff
@@ -0,0 +1,238 @@
+use super::Bytes;
+use super::BytesError;
+use super::Uuid;
+use super::Variant;
+use super::Version;
+
+/// A builder struct for creating a [`Uuid`]
+#[allow(missing_copy_implementations)]
+#[derive(Debug)]
+pub struct UuidBuilder {
```

Comment:
maybe make this a `newtype` tuple? Just to make it consistent with the rest of the crate